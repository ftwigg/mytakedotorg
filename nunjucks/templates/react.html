{% extends "reactCore.html" %}

{% block bundle %}
	<!-- Main -->
	<script src="/app.bundle.js"></script>
{% endblock %}
