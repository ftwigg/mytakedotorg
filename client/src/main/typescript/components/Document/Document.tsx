import * as React from "react";
import * as ReactDOM from "react-dom";
import isEqual = require("lodash/isEqual");
import {
  getNodeArray,
  getCaptionNodeArray,
  FoundationNode
} from "../../utils/functions";
import { CaptionWord, CaptionMeta } from "../../utils/databaseData";
import DocumentTextNodeList from "../DocumentTextNodeList";
import CaptionTextNodeList from "../CaptionTextNodeList";

interface CaptionData {
  captionTimer: number;
  captionMap: CaptionWord[];
  captionMeta: CaptionMeta;
}

interface DocumentProps {
  onMouseUp: () => void;
  nodes: FoundationNode[];
  className?: string;
  captionData?: CaptionData;
}

interface DocumentState {}

class Document extends React.Component<DocumentProps, DocumentState> {
  constructor(props: DocumentProps) {
    super(props);
  }
  getDocumentNodes = () => {
    return [...this.props.nodes];
  };
  render() {
    let classes = "document document--static";
    let documentClass = this.props.className
      ? this.props.className
      : "document__row";

    let childComponent;
    if (this.props.captionData) {
      throw "TODO";
      // childComponent = (
      //   <CaptionTextNodeList
      //     className="document__text document__text--caption"
      //     onMouseUp={this.props.onMouseUp}
      //     documentNodes={this.props.documentNodes}
      //     captionTimer={this.props.captionData.captionTimer}
      //     captionWordMap={this.props.captionData.captionMap}
      //     captionMeta={this.props.captionData.captionMeta}
      //   />
      // );
    } else {
      childComponent = (
        <DocumentTextNodeList
          className="document__text"
          onMouseUp={this.props.onMouseUp}
          documentNodes={this.props.nodes}
        />
      );
    }

    return (
      <div className={classes}>
        <div className={documentClass}>
          <div className={"document__row-inner"}>
            {childComponent}
            {this.props.children ? this.props.children : null}
          </div>
        </div>
      </div>
    );
  }
}

export default Document;
