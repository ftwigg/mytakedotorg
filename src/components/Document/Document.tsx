import * as React from "react";
import * as ReactDOM from "react-dom";
import { FoundationNode } from "../Foundation";
import { getNodeArray } from "../../utils/functions";

interface DocumentProps {
  onMouseUp: () => void;
  excerptId: string;
}

interface DocumentState {
  documentNodes: FoundationNode[];
}

class Document extends React.Component<DocumentProps, DocumentState> {
  constructor(props: DocumentProps) {
    super(props);

    this.state = {
      documentNodes: getNodeArray(props.excerptId)
    };
  }
  getDocumentNodes = () => {
    return [...this.state.documentNodes];
  };
  componentWillReceiveProps(nextProps: DocumentProps) {
    if (this.props.excerptId !== nextProps.excerptId) {
      this.setState({
        documentNodes: getNodeArray(nextProps.excerptId)
      });
    }
  }
  render() {
    let classes = "document document--static";
    let documentClass = "document__row";

    return (
      <div className={classes}>
        <div className={documentClass}>
          <div className={"document__row-inner"}>
            <div className={"document__text"} onMouseUp={this.props.onMouseUp}>
              {this.state.documentNodes.map(function(
                element: FoundationNode,
                index: number
              ) {
                element.props["key"] = index.toString();
                return React.createElement(
                  element.component,
                  element.props,
                  element.innerHTML
                );
              })}
            </div>
            {this.props.children ? this.props.children : null}
          </div>
        </div>
      </div>
    );
  }
}

export default Document;
