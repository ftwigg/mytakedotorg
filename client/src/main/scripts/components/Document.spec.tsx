import * as React from "react";
import * as renderer from "react-test-renderer";
import {} from "jest";
import Document from "./Document";

const nodes = [
  {
    component: "p",
    innerHTML: [
      "Section 1. Neither slavery nor involuntary servitude, except as a punishment for crime whereof the party shall have been duly convicted, shall exist within the United States, or any place subject to their jurisdiction."
    ],
    offset: 0
  },
  {
    component: "p",
    innerHTML: [
      "Section 2. Congress shall have power to enforce this article by appropriate legislation."
    ],
    offset: 218
  }
];
const className = "document__row";
const onMouseUp = jest.fn();

test("Document component", () => {
  const tree = renderer
    .create(
      <Document nodes={nodes} className={className} onMouseUp={onMouseUp} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

const children = (
  <div className="editor__block editor__block--overlay" style={{ top: "16px" }}>
    <div className="editor__document editor__document--hover">
      <p data-char-offset="0">
        Section 1.{" "}
        <span className="document__text--selected">
          Neither slavery nor involuntary servitude
        </span>, except as a punishment for crime whereof the party shall have
        been duly convicted, shall exist within the United States, or any place
        subject to their jurisdiction.
      </p>
    </div>
  </div>
);

test("Document component with highlights", () => {
  const tree = renderer
    .create(
      <Document nodes={nodes} className={className} onMouseUp={onMouseUp}>
        {children}
      </Document>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
