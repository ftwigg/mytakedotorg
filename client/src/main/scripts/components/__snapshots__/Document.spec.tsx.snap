// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document component 1`] = `
<div
  className="document document--static"
>
  <div
    className="document__row"
  >
    <div
      className="document__row-inner"
    >
      <div
        className="document__text"
        onMouseUp={[Function]}
      >
        <p
          data-char-offset={0}
        >
          Section 1. Neither slavery nor involuntary servitude, except as a punishment for crime whereof the party shall have been duly convicted, shall exist within the United States, or any place subject to their jurisdiction.
        </p>
        <p
          data-char-offset={218}
        >
          Section 2. Congress shall have power to enforce this article by appropriate legislation.
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Document component with highlights 1`] = `
<div
  className="document document--static"
>
  <div
    className="document__row"
  >
    <div
      className="document__row-inner"
    >
      <div
        className="document__text"
        onMouseUp={[Function]}
      >
        <p
          data-char-offset={0}
        >
          Section 1. Neither slavery nor involuntary servitude, except as a punishment for crime whereof the party shall have been duly convicted, shall exist within the United States, or any place subject to their jurisdiction.
        </p>
        <p
          data-char-offset={218}
        >
          Section 2. Congress shall have power to enforce this article by appropriate legislation.
        </p>
      </div>
      <div
        className="editor__block editor__block--overlay"
        style={
          Object {
            "top": "16px",
          }
        }
      >
        <div
          className="editor__document editor__document--hover"
        >
          <p
            data-char-offset="0"
          >
            Section 1. 
            <span
              className="document__text--selected"
            >
              Neither slavery nor involuntary servitude
            </span>
            , except as a punishment for crime whereof the party shall have been duly convicted, shall exist within the United States, or any place subject to their jurisdiction.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
